### PR TITLE
[YLP-702] update logic by returning a new token with updated user data

### PR DESCRIPTION
**Pull request by Nas84**

### CODE REVIEW

File: user/api.go [hunk 1]
```diff
@@ -539,8 +539,20 @@ func (a *Api) UpdateUser(res http.ResponseWriter, req *http.Request, vars map[st
...
-			a.logAudit(req, tokenData, "UpdateUser isClinic{%t}", updatedUser.IsClinic())
-			a.sendUser(res, updatedUser, tokenData.IsServer)
+			// Generate new token after update done in db (could be factorize with create User func)
+			role := "patient"
+			if updatedUser.Roles != nil && len(updatedUser.Roles) > 0 {
+				role = updatedUser.Roles[0]
+			}
+			newTokenData := token.TokenData{DurationSecs: extractTokenDuration(req), UserId: updatedUser.Id, IsServer: tokenData.IsServer, Role: role}
+			tokenConfig := token.TokenConfig{DurationSecs: a.ApiConfig.TokenDurationSecs, Secret: a.ApiConfig.Secret}
+			if sessionToken, err := CreateSessionTokenAndSave(req.Context(), &newTokenData, tokenConfig, a.Store); err != nil {
```
Comment (ydrisR) [inline user/api.go]: This changes the behaviour of the api and it does not seems correct, updateUser should not return a new token.

**Review by ydrisR — CHANGES_REQUESTED** [review]
I don't think we should modify the behaviour of the api "Update User". This endpoint should not return a new token. With the proposed change a service account (server token) will receive an "inconsistent" token where userId is a real user id (the target user we want to update), the role is "HCP" and isServer = true.

I would rather modify the method `RefreshSession` (` [GET] /login`) to return a new "correct" token. And at the same time it would be interesting to "invalidate" or remove the previous token.

File: user/api.go (hunk 1 above)
Comment (Nas84) [reply user/api.go]: we decide to perform it here as it more consistent wit the other api /login  and "/createuser". However you were not in the call we could discuss it more of course

Comment (Nas84) [reply user/api.go]: As discussed, indeed the token return by the "/create user" is somehow invalid and not useable. The best solution is to modify the method _RefreshSession_ (` [GET] /login`) to return a new "correct" token.

File: user/api.go [hunk 2]
```diff
@@ -814,7 +814,7 @@ func (a *Api) ServerLogin(res http.ResponseWriter, req *http.Request) {
 // @Failure 401 {string} string ""
 // @Router /login [get]
 func (a *Api) RefreshSession(res http.ResponseWriter, req *http.Request) {
-
+	a.logger.Printf("refresh session token %v", req)
```
Comment (yvon-dblg) [inline user/api.go]: This log is good for debug & prod.
But maybe the request content leak too much in the log?

Comment (Nas84) [reply user/api.go]: yep agree, i removed it and add trace token as discussed 😉

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ Shoreline is the module that manages logins and user accounts.
 ## 1.5.1 - 2021-05-03
 ### Fixed
 - Correct client mock so it returns the correct role
+- YLP-702: tokens do not contain relevant role
```
Comment (ydrisR) [inline CHANGELOG.md:8]: should be 1.5.2

File: user/api.go [hunk 4]
```diff
@@ -814,7 +814,7 @@ func (a *Api) ServerLogin(res http.ResponseWriter, req *http.Request) {
 // @Failure 401 {string} string ""
 // @Router /login [get]
 func (a *Api) RefreshSession(res http.ResponseWriter, req *http.Request) {
-
+	a.logger.Printf("refresh session token %v and trace %v", req.Header.Get(TP_SESSION_TOKEN), req.Header.Get(TP_TRACE_SESSION))
```
Comment (ydrisR) [inline user/api.go]: Need to print the session token?

**Review by ydrisR — CHANGES_REQUESTED** [review]

**Review by ydrisR — APPROVED** [review]
LGTM

**Pull request merged**